### PR TITLE
firmware: dfu: add dfu documentation

### PR DESCRIPTION
**Pull request by szczys**
Add DFU documentation for the Firmware SDK.

### CODE REVIEW

File: docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Golioth Firmware SDK includes DFU sample code which runs on all Continuously
+Verified Boards. A walkthrough is included here to help you become familiar with
+the process.
+
+## OTA Overview
+
+### Device Side
+
+The firmware update process is based around the open source
+[MCUboot](https://www.mcuboot.com/) project. MCUboot uses multiple flash memory
```
Comment (sam-golioth) [inline docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md:18]: Is this true for all ports?

Comment (szczys) [reply docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md:18]: Hmmm, probably not for ESP-IDF. Do you have any advice on how to handle this portion of the documentation in an abstract way?

Comment (sam-golioth) [reply docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md:18]: I don't know that we need to describe what happens on the device in much detail. The service we provide is for getting the right artifact from the cloud to the device. What the device does with that artifact (i.e. how the update is applied) is an exercise for the user (with reference implementations for our standard ports). I know that right now the update is largely under the control of the SDK, but we've talked about moving that out to the app like in the Zephyr SDK, and so I think it's ok for the docs to move in that direction and be a little bit more vague about what happens on the device.

Comment (szczys) [reply docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md]: @sam-golioth that helps clarify, thanks!

I respun that paragraph, please let me know what you think.

File: docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+release version and begin the update process. Golioth includes a one-click
+rollback feature if you ever need to revert a rollout.
+
+### Device Side
+
+When a device registers for firmware upgrades, it will receive a manifest from
+the Golioth cloud each time a release rollout change occurs. The SDK
+automatically parses the manifest, comparing the received firmware version
+number with the version the device is currently running. If the manifest version
+is newer, the SDK begins block download of the artifact. The exact device
```
Comment (sam-golioth) [inline docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md:31]:
```suggestion
is newer, the SDK begins a block download of the artifact. The exact device
```

**Review by sam-golioth — APPROVED** [review]

File: docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md (hunk 1 above)
Comment (sam-golioth) [reply docs/firmware/golioth-firmware-sdk/firmware-upgrade/README.md]: looks good to me!

**Pull request merged**